GoogleForCreators/web-stories-wp
# Editor: Display dialog if lacking upload permissions

Adds a dialog to display the access restriction warning message.

## Summary

Add a popup dialog to notify the user to check access permission when they are restricted to upload.

## Relevant Technical Choices

1. Add a RestrictAccessDialog component
2. Add a state [hasRestrictAccess] and pass MissingUploadPermissionDialog component under mediaPane component.
3. Add a onRestrictAccess props function on useMediaPicker, for controlling the dialog show.

## To-do

N/A

## User-facing changes

Will get a popup warning dialog instead of Snackbar when they click on the upload button and do not have access.

## Testing Instructions

1. The Snackbar should no longer shows up.
2. The dialog should popup with the accurate instruction test when not permitted user click on the 'Upload' button.
3. The dialog should close when the user click on 'GOT IT'

---

Fixes #2481
